kore/src/Kore/Step/Simplification/Not.hs

```diff
@@ -77,31 +99,37 @@ besides the pattern sort, which will make it even more useful to carry around.
 
 -}
 simplifyEvaluated
-    ::  ( MetaOrObject level
+    ::  ( FreshVariable variable
         , SortedVariable variable
-        , Ord (variable level)
-        , Show (variable level)
-        , Unparse (variable level)
+        , Ord (variable Object)
+        , Show (variable Object)
+        , Unparse (variable Object)
         )
-    => OrOfExpandedPattern level variable
-    -> (OrOfExpandedPattern level variable, SimplificationProof level)
-simplifyEvaluated simplified
+    => MetadataTools Object Attribute.Symbol
+    -> PredicateSubstitutionSimplifier Object
+    -> StepPatternSimplifier Object
+    -> BuiltinAndAxiomSimplifierMap Object
+    -> OrOfExpandedPattern Object variable
+    -> BranchT Simplifier (ExpandedPattern Object variable)
+simplifyEvaluated
+    tools
+    predicateSimplifier
+    termSimplifier
+    axiomSimplifiers
+    simplified
   | OrOfExpandedPattern.isFalse simplified =
-    (MultiOr.make [ExpandedPattern.top], SimplificationProof)
+    return ExpandedPattern.top
   | OrOfExpandedPattern.isTrue simplified =
-    (MultiOr.make [], SimplificationProof)
+    empty
   | otherwise =
-    case MultiOr.extractPatterns simplified of
-        [patt] -> makeEvaluate patt
-        _ ->
-            ( makeFromSinglePurePattern
-                (mkNot
-                    (ExpandedPattern.toMLPattern
-                        (OrOfExpandedPattern.toExpandedPattern simplified)
-                    )
-                )
-            , SimplificationProof
-            )
+    Foldable.foldrM mkAnd ExpandedPattern.top (simplified >>= makeEvaluate)
```

Comment:
For some reason, I don't find the MultiOr monad intuitive enough... Oh well, I guess I need to get used to it.